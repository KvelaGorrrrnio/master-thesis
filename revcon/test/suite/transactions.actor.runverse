test/suite/transactions.actor.revcon:
[store]
  !mailbox: [>MessageVal (-1) 0 "init" []>]
  !message: Ø
  !messages: []
  main: Ø

[stack]
[spawn main main | from UnOpEmpty (!mailbox) do 
[pop !message !messages | undeliver "!message" | undequeue !message !mailbox] <|> []
 loop 
[] <|> []
 until UnOpEmpty (!messages) | unsend Left "main".init() | despawn main main] <|> []


[procedures]
bank.close()
bank.deposit(amount, account)
bank.open(inVault)
bank.withdraw(amount, account)
main.closed(inVault)
main.deposited(inAmount, inAccount)
main.init()
main.opened()
main.waitForRequests()
main.withdrawn(inAmount, inAccount)

[globals]
bad_requests: 0
vault: []
wallet: 0
