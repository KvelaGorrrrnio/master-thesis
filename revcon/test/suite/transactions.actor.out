test/suite/transactions.actor.revcon:
[store]
  !mailbox: [>>]
  !message: Ø
  !messages: [MessageVal 1 0 "closed" [[]], MessageVal 0 1 "close" [], MessageVal 0 0 "waitForRequests" [], MessageVal 1 0 "deposited" [7213,2], MessageVal 1 0 "withdrawn" [3500,5], MessageVal 1 0 "withdrawn" [0,2], MessageVal 1 0 "deposited" [5000,5], MessageVal 0 0 "waitForRequests" [], MessageVal 0 1 "deposit" [0,0], MessageVal 0 1 "withdraw" [0,0], MessageVal 0 1 "withdraw" [7000,0], MessageVal 0 1 "deposit" [0,0], MessageVal 1 0 "opened" [], MessageVal 0 1 "open" [[]], MessageVal (-1) 0 "init" []]
  main: Ø

[stack]
[spawn main main | send Left "main".init() | from UnOpEmpty (!messages) do 
[dequeue !message !mailbox | deliver "!message" | push !message !messages] <|> []
 loop 
[] <|> []
 until UnOpEmpty (!mailbox) | despawn main main] <|> []


[procedures]
bank.close()
bank.deposit(amount, account)
bank.open(inVault)
bank.withdraw(amount, account)
main.closed(inVault)
main.deposited(inAmount, inAccount)
main.init()
main.opened()
main.waitForRequests()
main.withdrawn(inAmount, inAccount)

[globals]
bad_requests: 1
vault: [0, 0, 7213, 0, 0, 1500, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
wallet: 91287
